lib/screenplay_web/controllers/config_controller.ex

```diff
@@ -38,4 +42,46 @@ defmodule ScreenplayWeb.ConfigController do
         send_resp(conn, 400, "S3 Operation Failed: Put")
     end
   end
+
+  def existing_screens(conn, %{"place_ids" => place_ids, "app_id" => app_id}) do
+    app_id_atom = String.to_existing_atom(app_id)
+
+    places_and_screens =
+      place_ids
+      |> String.split(",")
+      |> Enum.map(fn place_id ->
+        filter_fn = fn
+          {_, %Screen{app_id: ^app_id_atom} = config} ->
+            place_id_has_screen?(place_id, String.to_existing_atom(app_id), config)
+
+          _ ->
+            false
+        end
+
+        live_screens =
+          ScreensConfigCache.screens(&filter_fn.(&1))
+          |> Enum.map(fn screen ->
+            Map.put(screen, :is_live, true)
```

Comment:
Same here, it's best not to add arbitrary new fields to a struct.

If you need to associate the config with some extra info, you could define a separate struct that holds a screen config, along with that extra info.
Or you could go the other way and do `Map.from_struct(screen)` earlier on to switch to working with a plain map.